test/exercises/diffusion/heat_2d_dist_block.chpl

```diff
@@ -0,0 +1,57 @@
+/*
+  A distributed 2D finite-difference heat/diffusion equation solver
+
+  Computation is executed over a 2D distributed array.
+  The array distribution is managed by the `Block` distribution.
+  The `forall` loop manages task creation and synchronization
+  across and within locales.
+*/
+
+import BlockDist.Block,
+       Time.stopwatch;
+
+// create a stopwatch to time kernel execution
+var t = new stopwatch();
+config const writeTime = false;
+
+// declare configurable constants with default values
+config const nx = 256,      // number of grid points in x
+             ny = 256,      // number of grid points in y
+             nt = 50,       // number of time steps
+             alpha = 0.25;  // diffusion constant
+
+// define a distributed 2D domain and subdomain to describe the grid and its interior
+const indices = {0..<nx, 0..<ny},
+      indicesInner = indices.expand(-1),
+      INDICES = Block.createDomain(indices),
+      INDICES_INNER = INDICES[indicesInner];
+
+// define a distributed 2D array over the above domain
+var u: [INDICES] real;
+
+// set up initial conditions
+u = 1.0;
+u[nx/4..nx/2, ny/4..ny/2] = 2.0;
+
+// create a temporary copy of 'u' to store the previous time step
+var un = u;
```

Comment:
In the mathematical notation, `u` would typically be a lower case letter since it's a scalar quantity. I feel pretty inclined to keep it that way for a scientific-computing audience. Let me know if you feel strongly enough to overwrite that.

Reply:
scalar in what sense?  That each element of the array is a scalar?  Or something else?

(to my thinking, `u[i,j]` is a scalar, and `U` is the collection of all scalars, such that I might write `ref u_ij = U[i,j];` or `forall u in U do ...`.  But I assume you're saying that, mathematically speaking, `u` would be used for the collection of all scalars as well?)

I'm definitely fine deferring to the math if that's going to make it more attractive to readers from the community in question.  I.e., I wouldn't go with my preferred naming scheme over one that will make the code most attractive and approachable to a given community.

Reply:
>  But I assume you're saying that, mathematically speaking, u would be used for the collection of all scalars as well?

Yeah, essentially `u` represents a scalar value at each point in space (density of the fluid or temperature of a material), so it would be considered a scalar quantity as a whole.  Alternatively a vector quantity, for example the [electric field](https://en.wikipedia.org/wiki/Electric_field), represents a vector at each point in space, so it typically uses an uppercase letter.